File: Atcoder/ARC110A.py
import math

# 最小公倍数
def lcm(x, y):
    return (x * y) // math.gcd(x, y)

N = int(input())
answer = 1

for i in range(2, N+1):
    answer = lcm(answer, i)

print(answer + 1)

File: Atcoder/ARC110A.cpp
#include<bits/stdc++.h>

using namespace std;

// 最大公約数
long long GCD(long long a, long long b) {
    if (b == 0) return a;
    else return GCD(b, a % b);
}

// 最小公倍数
long long LCM(long long x, long long y) {
	return x * y / GCD(x, y);
}

int main() {
	long long N;
	cin >> N;

	long long tmp = 1;
	
	for (long long i = 2; i <= N; i++) {
		tmp = LCM(tmp, i);
	}

	cout << tmp+1 << endl;
	return 0;
}